### DISCUSSION TITLE
heads up: Moka v0.7.0 has a breaking change

### DISCUSSION THREAD

**Discussion by tatsuya6502** [messense/aliyundrive-webdav#231] [General]
Hi @messense,

Just a heads up. I have released Moka [v0.7.0](https://crates.io/crates/moka/0.7.0) but it has a small breaking change. I had to change the type of `max_capacity` from `usize` to `u64` to support a new feature. So when you update the dependency for Moka, you will have to change the `max_capacity` in the following code to `u64` too.

https://github.com/messense/aliyundrive-webdav/blob/bc801e1d8eac63e063fd967a52e0532b1a6bcf63/src/cache.rs#L14-L20

FYI, now Moka supports size-aware eviction. Instead of bounding a cache with the max number of cache entries, you can bound a cache with a sum of physical (or logical) sizes of entries. Here is [a link](https://github.com/moka-rs/moka/blob/609ea8ff3354e09664335864afc095dcd3790618/README.md#example-size-aware-eviction) to an example.

Also note that, in the example, a new method `builder()` is used. Now you can write `Cache::builder().max_capacity(x) ... build()` instead of `CacheBuilder::new(x) ... build()`. It is a small change but you can remove `CacheBuilder` from `use` statement.

**Comment by messense** [comment]
Thanks! Upgraded moka in https://github.com/messense/aliyundrive-webdav/commit/9339cf8c458a8dc5c3ff1a8bab34152752932e7e
